src/repository/TreeRepository.ts

```diff
@@ -265,7 +265,7 @@ export class TreeRepository<Entity> extends Repository<Entity> {
         const parentEntityId = this.metadata.primaryColumns[0].getEntityValue(entity);
         const childRelationMaps = relationMaps.filter(relationMap => relationMap.parentId === parentEntityId);
         const childIds = new Set(childRelationMaps.map(relationMap => relationMap.id));
-        entity[childProperty] = entities.filter(entity => childIds.has(entity.id));
+        entity[childProperty] = entities.filter(entity => childIds.has(entity[this.metadata.primaryColumns[0].propertyName]));
```

Comment:
relying on a single primary column always means there would be issues with multiple primary columns...

Reply:
Hundred percent agree.

The reason why I ended up accepting this is that there's several other places in the same file where this pattern is used - using the first primary column only - without even checking for its existence.

If we were to support multiple primary keys here it would have meant refactoring how other parts of the Tree Repository works, and balloon the scope of the change dramatically.